extensions/sql-database-projects/src/models/project.ts

```diff
@@ -306,80 +306,63 @@ export class Project implements ISqlProject {
-	 *	item already exists in the project file, then existing entry will be returned.
 	 */
-	public async addFolderItem(relativeFolderPath: string, doNotThrowOnDuplicate?: boolean): Promise<FileProjectEntry> {
-		const absoluteFolderPath = path.join(this.projectFolderPath, relativeFolderPath);
-		const normalizedRelativeFolderPath = utils.convertSlashesForSqlProj(relativeFolderPath);
+	public async addFolderItem(relativeFolderPath: string): Promise<FileProjectEntry> {
+		const folderEntry = await this.ensureFolderItems(relativeFolderPath);
 
-		// check if folder already has been added to sqlproj
-		const existingEntry = this.files.find(f => f.relativePath.toUpperCase() === normalizedRelativeFolderPath.toUpperCase());
-		if (existingEntry) {
-			if (!doNotThrowOnDuplicate) {
-				throw new Error(constants.folderAlreadyAddedToProject((relativeFolderPath)));
-			}
-
-			return existingEntry;
-		}
-
-		// If folder doesn't exist, create it
-		let exists = await utils.exists(absoluteFolderPath);
-		if (!exists) {
-			await fs.mkdir(absoluteFolderPath, { recursive: true });
+		if (folderEntry) {
+			return folderEntry;
+		} else {
+			throw new Error(constants.outsideFolderPath);
 		}
-
-		const folderEntry = this.createFileProjectEntry(normalizedRelativeFolderPath, EntryType.Folder);
-		this._files.push(folderEntry);
-
-		await this.addToProjFile(folderEntry);
-		return folderEntry;
 	}
 
 	/**
 	 * Writes a file to disk if contents are provided, adds that file to the project, and writes it to disk
+	 *
 	 * @param relativeFilePath Relative path of the file
 	 * @param contents Contents to be written to the new file
 	 * @param itemType Type of the project entry to add. This maps to the build action for the item.
-	 * @param doNotThrowOnDuplicate
-	 *	Flag that indicates whether duplicate entries should be ignored or throw an error. If flag is set to `true` and
-	 *	item already exists in the project file, then existing entry will be returned.
 	 */
-	public async addScriptItem(relativeFilePath: string, contents?: string, itemType?: string, doNotThrowOnDuplicate?: boolean): Promise<FileProjectEntry> {
+	public async addScriptItem(relativeFilePath: string, contents?: string, itemType?: string): Promise<FileProjectEntry> {
 		const absoluteFilePath = path.join(this.projectFolderPath, relativeFilePath);
 
-		// check if file already exists if content was passed to write to a new file
-		if (contents !== undefined && contents !== '' && await utils.exists(absoluteFilePath)) {
-			throw new Error(constants.fileAlreadyExists(path.parse(absoluteFilePath).name));
+		if (contents) {
+			// Create the file if contents were passed in and file does not exist yet
+			await fs.mkdir(path.dirname(absoluteFilePath), { recursive: true });
+
+			try {
+				await fs.writeFile(absoluteFilePath, contents, { flag: 'wx' });
+			} catch (error) {
+				if (error.code === 'EEXIST') {
```

Comment:
Updated the code here to actually ensure that files are not overwritten, as that seems to have been the intention of the separate `utils.exists(..)` check.